### ISSUE TITLE
[BUG] The exported PFX certificate doesn't have the full Let's Encrypt chain

### ISSUE THREAD

**Issue by tallen116** [PKISharp/ACME-PS#140]
**Describe the bug**
When exporting using `Export-ACMECertificate` command, it is missing the Let’s Encrypt R3 intermediate certificate.

**To Reproduce**

```
$acmeStateDir = "path to acme state directory"
$acmeServiceName = "LetsEncrypt-Staging"
$contactMailAddresses = @("email1", "email2")
$dnsIdentifiers = @("dns1")
$certExportPath = "path to export"

New-ACMEState -Path $acmeStateDir
Get-ACMEServiceDirectory -State $acmeStateDir -ServiceName $acmeServiceName
New-ACMEAccountKey -State $acmeStateDir
New-ACMEAccount -State $acmeStateDir -EmailAddresses $contactMailAddresses -AcceptTOS

$order = New-ACMEOrder -State $acmeStateDir -Identifiers $dnsIdentifiers
$authorizations = @(Get-ACMEAuthorization -State $acmeStateDir -Order $order)

$authZ = $authorizations[0]
$challenge = Get-ACMEChallenge -State $acmeStateDir -Authorization $authZ -Type "http-01"
# Upload token to web server
$challenge | Complete-ACMEChallenge -State $acmeStateDir
$order | Update-ACMEOrder -State $acmeStateDir -PassThru
Complete-ACMEOrder -State $acmeStateDir -Order $order -GenerateCertificateKey
$securePassword = ConvertTo-SecureString "<password>" –asplaintext –force

Export-ACMECertificate -State $acmeStateDir `
        -Order $order `
        -Path $certExportPath `
        -Password $securePassword
```

Using openssl to check the certificate and it only has 2 certs and a key.  It seems to be missing the `Let’s Encrypt R3 (RSA 2048, O = Let's Encrypt, CN = R3` certificate.  

https://letsencrypt.org/certificates/

```
openssl pkcs12 -info -in certificate_1.pfx -nodes | grep -E 'subject=|issuer='

subject=CN = <website dns name>
issuer=C = US, O = Let's Encrypt, CN = R3
subject=C = US, O = Internet Security Research Group, CN = ISRG Root X1
issuer=O = Digital Signature Trust Co., CN = DST Root CA X3
```

I see the downloaded certificate in the state directory has the full chain.  I did try to debug some of the code in `https://github.com/PKISharp/ACME-PS/blob/master/ACME-PS/functions/Certificate/Export-Certificate.ps1` and it did seem to parse out the 3 certs before calling the final `$certContent = [Certificate]::ExportPfxCertificateChain($certificates, $CertificateKey, $Password)`.

**Expected behavior**
It should provide all chain certificates when exporting to PFX

**Additional context (Powershell version, your environment, etc.)**
ACME-PS: 1.5.8
Powershell: 5.1

**Comment by tallen116** [comment]
Just looking at some things and I think the logic around this might be wrong.

File: `https://github.com/PKISharp/ACME-PS/blob/master/ACME-PS/internal/classes/crypto/Certificate.ps1`
Line: 69

The for loop looks like it wants to loop through the `byte[][]` converted `ArrayList` getting all certs except the first one.  *Assuming the server sends the certificates in the correct order

It looks like the `$i -gt 1` might be a mistake if my assumptions are correct.  Here is some test code just to vet out the current code.

```
PS #> $certs = [System.Collections.ArrayList]::new()
PS #> $certs.add("1")                               
0
PS #> $certs.add("2")                               
1
PS #> $certs.add("3")
2
PS #> $certs
1
2
3
PS #> $certByte = [byte[][]]$certs                  
PS #> for($i = $certByte.Length - 1; $i -gt 1; $i--) {
>> Write-Output $certByte[$i]
>> }
3
PS #> 
```

It will miss the next to last one.  Changing it to either `$i -gt 0` or `$i -ge 1` gets all of them except the first.

```
PS #> for($i = $certByte.Length - 1; $i -gt 0; $i--) {
>> Write-Output $certByte[$i]
>> }
3
2
PS #> for($i = $certByte.Length - 1; $i -ge 1; $i--) {
>> Write-Output $certByte[$i]
>> }
3
2
PS #> 
```

**Comment by glatzert** [comment]
Yay fell in one of the 2 deadly programmer traps:
- Naming things
- Cache invalidation
- off by one errors

:)

If you like provide a PR for a fix.
If not, I'll handle it somewhen this week.

**Comment by tallen116** [comment]
I can provide a PR later.  Thank you for your quick response.

**Comment by glatzert** [comment]
Solved via #141
